specification/managementgroups/resource-manager/Microsoft.Management/stable/2020-02-01/management.JSON

```diff
@@ -0,0 +1,2000 @@
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The fully qualified ID for the child resource (management group or subscription).  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the child entity."
+        },
+        "displayName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The friendly name of the child resource."
+        },
+        "roles": {
+          "description": "The roles definitions associated with the management group.",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "children": {
+          "description": "The list of children.",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CreateManagementGroupChildInfo"
+          }
+        }
+      }
+    },
+    "CreateParentGroupInfo": {
+      "description": "(Optional) The ID of the parent management group used during creation.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified ID for the parent management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the parent management group"
+        },
+        "displayName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The friendly name of the parent management group."
+        }
+      }
+    },
+    "Permissions": {
+      "type": "string",
+      "enum": [
+        "noaccess",
+        "view",
+        "edit",
+        "delete"
+      ],
+      "description": "The users specific permissions to this item."
+    },
+    "CheckNameAvailabilityRequest": {
+      "description": "Management group name availability check parameters.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "the name to check for availability"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "/providers/Microsoft.Management/managementGroup"
+          ],
+          "x-ms-enum": {
+            "name": "Type",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "/providers/Microsoft.Management/managementGroup"
+              }
+            ]
+          },
+          "description": "fully qualified resource type which includes provider namespace"
+        }
+      }
+    },
+    "ProvisioningState": {
```

Comment:
I don't think this definition is used.

Reply:
removing

Reply:
All the resources must have provisioningState property, in the property bag . I am sure implementation has it.
This is required for template deployments.
Please add this field to all the resources.

---
In reply to: [388549550](https://github.com/Azure/azure-rest-api-specs/pull/8443#discussion_r388549550) [](ancestors = 388549550)

Reply:
This is not implemented on our side. We can add it as a bug in our code that prevents template deployments from working on MGs and we can fix it on future API versions.